test/ci-operator-integration/config/test-config.yaml

```diff
@@ -0,0 +1,246 @@
+      paths: null
+  to: hyperkube
+- dockerfile_path: images/tests/Dockerfile.rhel
+  from: cli
+  inputs:
+    bin:
+      as:
+      - registry.svc.ci.openshift.org/ocp/builder:golang-1.12
+      paths: null
+  to: tests
+- context_dir: images/os/
+  from: base
+  inputs:
+    base-machine-with-rpms:
+      as:
+      - fedora:29
+      paths: null
+    machine-os-content-base:
+      as:
+      - registry.svc.ci.openshift.org/openshift/origin-v4.0:machine-os-content
+      paths: null
+  to: machine-os-content
+promotion:
+  additional_images:
+    artifacts: artifacts
+  excluded_images:
+  - machine-os-content
+  name: "4.3"
+  namespace: ocp
+raw_steps:
+- pipeline_image_cache_step:
+    commands: mkdir -p _output/local/releases; touch _output/local/releases/CHECKSUM;
+      echo $'FROM bin AS bin\nFROM rpms AS rpms\nFROM centos:7\nCOPY --from=bin /go/src/github.com/openshift/origin/_output/local/releases
+      /srv/zips/\nCOPY --from=rpms /go/src/github.com/openshift/origin/_output/local/releases/rpms/*
+      /srv/repo/' > _output/local/releases/Dockerfile; make build-cross
+    from: bin
+    to: bin-cross
+- project_directory_image_build_step:
+    from: base
+    inputs:
+      bin-cross:
+        as:
+        - bin
+        paths:
+        - destination_dir: .
+          source_path: /go/src/github.com/openshift/origin/_output/local/releases/Dockerfile
+      rpms:
+        as:
+        - rpms
+        paths: null
+      src:
+        paths: null
+    optional: true
+    to: artifacts
+- output_image_tag_step:
+    from: artifacts
+    optional: true
+    to:
+      name: stable
+      namespace: ""
+      tag: artifacts
+- rpm_image_injection_step:
+    from: base
+    to: base-with-rpms
+- rpm_image_injection_step:
+    from: base-machine
+    to: base-machine-with-rpms
+resources:
+  '*':
+    limits:
+      memory: 8Gi
+    requests:
+      cpu: 100m
+      memory: 4Gi
+  bin:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 7Gi
+  bin-cross:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 8Gi
+  cmd:
+    limits:
+      memory: 11Gi
+    requests:
+      cpu: "4"
+      memory: 8Gi
+  hyperkube:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 7Gi
+  hypershift:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 7Gi
+  integration:
+    limits:
+      memory: 18Gi
+    requests:
+      cpu: "6"
+      memory: 14Gi
+  rpms:
+    limits:
+      memory: 10Gi
+    requests:
+      cpu: "4"
+      memory: 8Gi
+  tests:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 7Gi
+  unit:
+    limits:
+      memory: 11Gi
+    requests:
+      cpu: "6"
+      memory: 8Gi
+  verify:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 8Gi
+rpm_build_commands: make build-rpms
+tag_specification:
+  name: "ci-operator-test"
+  namespace: ci
```

Comment:
Is this created by our automation tools?

Reply:
Nope. I added this imagestream because it won't change or so. Any other suggestions are welcome.

Reply:
Yeah, the test being dependent on external state is not great, but no one has come up with a better idea. I was just commenting that we want that ImageStream tracked in `openshift/release`.

Reply:
That is a good follow-up. I need also to investigate the case that the imagestream will be empty and so on.

Reply:
So we have to maintain this imagetream. However, even if we add this `is` in release repo, it will never be going to be automated. Not to mention the image that we have to push there...

Reply:
Do we require *valid* values for a dry-run? Can't we just use

```
$ oc create imagestream --dry-run -o yaml test
$ oc create imagestreamtag --dry-run --from test -o yaml test:test
```

?

Reply:
We do actually required valid values. `ci-operator` even in a `dry-run` mode will actually get the images from the imagestream.

Reply:
But that's just text read from the kube object, isn't it? There is no requirement that the pull spec actually refer to real images.